tests/lava/lib/dl/slayer/block/test_cuba.py

```diff
@@ -25,27 +26,78 @@
 torch.manual_seed(seed)
 torch.cuda.manual_seed(seed)
 if verbose:
-    print(f'{seed=}')
+    print(f"{seed=}")
 
 if torch.cuda.is_available():
-    device = torch.device('cuda')
+    device = torch.device("cuda")
 else:
     if verbose:
-        print('CUDA is not available in the system. '
-              'Testing for CPU version only.')
-    device = torch.device('cpu')
+        print(
+            "CUDA is not available in the system. "
+            "Testing for CPU version only."
+        )
+    device = torch.device("cpu")
 
-tempdir = os.path.dirname(__file__) + '/temp'
+tempdir = os.path.dirname(__file__) + "/temp"
 os.makedirs(tempdir, exist_ok=True)
 
-neuron_param = {'threshold': 0.5,
-                'current_decay': 0.5,
-                'voltage_decay': 0.5}
+neuron_param = {"threshold": 0.5, "current_decay": 0.5, "voltage_decay": 0.5}
 
 
 class TestCUBA(unittest.TestCase):
     """Test CUBA blocks"""
 
+    def test_affine_block_hdf5_export_dynamics_false(self):
+        """Test affine block hdf5 export in dynamics=false mode."""
+        in_features = 10
+        out_features = 5
+
+        net = slayer.block.cuba.Affine(
+            neuron_params=neuron_param,
+            in_neurons=in_features,
+            out_neurons=out_features,
+            dynamics=False,
+            count_log=False,
+        )
+
+        # export slayer network
+        h = h5py.File(tempdir + "/cuba_affine_dynamics_false.net", "w")
+        net.export_hdf5(h.create_group("layer/0"))
+
+        # reload net from h5 and check if 'neuron' exists.
+        lava_net = netx.hdf5.Network(
+            net_config=tempdir + "/cuba_affine_dynamics_false.net"
+        )
+        layer = lava_net.layers[0]
+
+        assert 'neuron' not in layer.__dict__.keys()
+
+    def test_affine_block_hdf5_export_dynamics_true(self):
+        """Test affine block hdf5 export in dynamics=true mode."""
+        in_features = 10
+        out_features = 5
+
+        net = slayer.block.cuba.Affine(
+            neuron_params=neuron_param,
+            in_neurons=in_features,
+            out_neurons=out_features,
+            dynamics=True,
+            count_log=False,
+        )
+
+        # export slayer network
+        h = h5py.File(tempdir + "/cuba_affine_dynamics_true.net", "w")
+        net.export_hdf5(h.create_group("layer/0"))
+
+        # reload net from h5 and check if 'vThMant' is '(1 << 17)'.
+        lava_net = netx.hdf5.Network(
+            net_config=tempdir + "/cuba_affine_dynamics_true.net"
+        )
+        layer = lava_net.layers[0]
+        neuron = layer.__dict__["neuron"].__dict__
+
+        assert neuron["vThMant"] == (1 << 18) - 1
```

Comment:
You can keep this one as well, but use `self.assertTrue(...)`

Reply:
@bamsumit the test for `False` passes now. I changed it to `'neuron' in lava_net.net_config['layer'][0].keys()` though.

For `True`, the threshold causes trouble because it is of type `None`, which is set in` src/lava/lib/dl/slayer/block/cuba.py class Affine`. I changed it to `-1` for `self.dynamics == True`, I hope this value is the correct one for non-spiking behavior. 

However, `'vThMant'` is not exported for some reason and the test fails. Have you an idea here?

Reply:
@ahenkes1 did you add this piece of code? https://github.com/lava-nc/lava-dl/pull/222#discussion_r1297817864 to export when `dynamics==True` ?

Reply:
@bamsumit yes, I added it to `block/base.py` (l.394)!

Reply:
@bamsumit the value is correctly exported to the h5 file, which can be checked by

```
        with h5py.File(tempdir + "/cuba_affine_dynamics_true.net", 'r') as hf:
            vThMant = np.array(hf['layer']['0']['neuron']['vThMant'])

        print(vThMant)
        print(vThMant.shape)
        print(vThMant.dtype)
        exit()
```
which results in the following output

> 262143
> ()
> int64

But it does not seem to appear in the exported neuron. Is this even necessary or should we simply compare with the h5 file?

Reply:
We can just compare the hdf5 file field for now.